src/components/legend/draw.js

```diff
@@ -25,31 +25,43 @@ var helpers = require('./helpers');
 var MAIN_TITLE = 1;
 
 module.exports = function draw(gd, opts) {
-    if(!opts) opts = gd._fullLayout.legend || {};
-    return _draw(gd, opts);
+    if(opts) {
+        drawOne(gd, opts);
+    } else {
+        var legends = gd._fullLayout._legends;
+        for(var i = 0; i < legends.length; i++) {
+            var legendId = legends[i];
+            var legendObj = gd._fullLayout[legendId];
+            drawOne(gd, legendObj);
+        }
+    }
 };
 
-function _draw(gd, legendObj) {
+function drawOne(gd, opts) {
+    var legendObj = opts || {};
+
     var fullLayout = gd._fullLayout;
-    var clipId = 'legend' + fullLayout._uid;
-    var layer;
+    var legendId = getId(legendObj);
+
+    var clipId, layer;
 
     var inHover = legendObj._inHover;
     if(inHover) {
         layer = legendObj.layer;
-        clipId += '-hover';
+        clipId = 'hover';
     } else {
         layer = fullLayout._infolayer;
+        clipId = legendId;
```

Comment:
Doesn't `clipId` need to include `fullLayout._uid` so it's unique page-wide if there are multiple plots with legends?

Reply:
Oh nvm you added that below!